### PR TITLE
Build and test with CUDA 12.5.1

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
This PR updates the latest CUDA build/test version 12.2.2 to 12.5.1.

Contributes to https://github.com/rapidsai/build-planning/issues/73

### CODE REVIEW

**Review by jameslamb — CHANGES_REQUESTED** [review]
I found one other place that needs to be updated:

https://github.com/rapidsai/cudf/blob/a6de6cc23702ed71b80625f461a90e910a33642f/CONTRIBUTING.md#L107

Since the name of the conda env file checked into `conda/environments` is changing (`-122` to `-125`), mentions of it in docs need to change.

Many of the RAPIDS projects have similar lines in their docs that describe using the files in `conda/environments` for local development (although of the ones I've looked at today, most are still using the CUDA 11.8 env), so I think it'd be worth adding this to the `rapids-reviser` migration script so such things aren't missed the next time we do an update like this.

**how I found this**

I've been doing the following while reviewing these CUDA 12.5 PRs:

```shell
# look for other uses of CUDA 12.2
git grep -E '12\.2'

# look for references to the conda env files
git grep -E 'all_cuda'
git grep -E '\-122'
```

**Comment by KyleFromNVIDIA** [comment]
I've updated https://github.com/rapidsai/rapids-reviser/pull/42 to update conda environments and docs, updated this PR, and also updated https://github.com/rapidsai/cugraph/pull/4534.

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

File: .devcontainer/cuda12.2-conda/devcontainer.json
Comment (bdice) [inline .devcontainer/cuda12.2-conda/devcontainer.json:1]: @KyleFromNVIDIA This directory needs to be renamed from 12.2 to 12.5. Maybe we can get by with just one devcontainer per major version, and use only “11” and “12”? Perhaps that would be worth discussing.

Comment (jakirkham) [reply .devcontainer/cuda12.2-conda/devcontainer.json:1]: Thanks Bradley! 🙏

We missed this on review. Sorry about that

Have simply renamed in PR: https://github.com/rapidsai/cudf/pull/16293 (this is what we needed to do in a couple other places already)

We can discussing renaming separately. Would want to hear from Paul on that

**Pull request merged**